core/src/main/java/com/gentics/mesh/core/data/generic/AbstractMeshCoreVertex.java

```diff
@@ -167,6 +170,32 @@ public String getETag(InternalActionContext ac) {
 		keyBuilder.append(getUuid());
 		keyBuilder.append("-");
 		keyBuilder.append(ac.getUser().getPermissionInfo(this).getHash());
+
+		keyBuilder.append("fields:");
+		GenericParameters generic = ac.getGenericParameters();
+		FieldsSet fields = generic.getFields();
+		fields.forEach(keyBuilder::append);
+
+		/**
+		 * permissions (&roleUuid query parameter aware)
+		 * 
+		 * Permissions can change and thus must be included in the etag computation in order to invalidate the etag once the permissions change.
+		 */
+		String roleUuid = ac.getRolePermissionParameters().getRoleUuid();
+		if (!isEmpty(roleUuid)) {
+			Role role = mesh().boot().meshRoot().getRoleRoot().loadObjectByUuid(ac, roleUuid, READ_PERM);
+			if (role != null) {
+				Set<GraphPermission> permSet = role.getPermissions(this);
+				Set<String> humanNames = new HashSet<>();
+				for (GraphPermission permission : permSet) {
+					humanNames.add(permission.getRestPerm().getName());
+				}
+				String[] names = humanNames.toArray(new String[humanNames.size()]);
+				keyBuilder.append(Arrays.toString(names));
+			}
+
+		}
+
```

Comment:
Create a new abstract method to get the etag information for the concrete class. Call it here and combine it with the generic etag information. This guarantees that the generic information will not be forgotten.